java/vector/src/main/java/org/apache/arrow/vector/VectorUnloader.java

```diff
@@ -76,6 +97,10 @@ private void appendNodes(FieldVector vector, List<ArrowFieldNode> nodes, List<Ar
           "wrong number of buffers for field %s in vector %s. found: %s",
           vector.getField(), vector.getClass().getSimpleName(), fieldBuffers));
     }
+
+    fieldBuffers = fieldBuffers.stream().map(
```

Comment:
small nit: if we aren't using streams elsewhere we should avoid using them here as well.  they can add some unncessary overhead for serialization/deserialization path.

Reply:
Agreed. I have changed it to a for loop.